convex/player.ts

```diff
@@ -0,0 +1,55 @@
+import { query, mutation } from "./_generated/server";
+import { v } from "convex/values";
+
+// start a game
+export const startGames = mutation({
+  args: { topic: v.string(), words: v.string(), roomId: v.id("rooms") },
+  handler: async (ctx, args) => {
+    const gameId = await ctx.db.insert("games", {
+      topic: args.topic,
+      words: args.words,
+    });
+    // update the player who in this room
+    const playerList = await ctx.db
+      .query("player")
+      .filter((q) => q.eq(q.field("roomId"), args.roomId))
+      .collect();
+    for (let i = 0; i < playerList.length; i++) {
+      const player = playerList[i];
+      if (player === undefined) {
+        console.error("The data error");
+      } else {
+        await ctx.db.patch(player._id, { gameId: gameId, score: 0 });
+      }
+    }
+  },
+});
+
+// check the user's word is correct or not
+export const checkWordFromUser = mutation({
+  args: {
+    playerId: v.id("player"),
+    wordFromUser: v.string(),
+    gameId: v.id("games"),
+  },
+  handler: async (ctx, args) => {
+    const games = await ctx.db.get(args.gameId);
+    // Update the score
+    if (games?.words?.includes(args.wordFromUser)) {
+      const player = await ctx.db.get(args.playerId);
+      const newScore = player?.score ? player?.score + 1 : 1;
+      ctx.db.patch(args.playerId, { score: newScore });
```

Comment:
You have to check if the user have not answered it yet. If you don't check it, I can spam any of the word and get unlimited score. Also, check that the word is not my word. I shouldn't be able to get score by sending my own word.

Reply:
yes, here can change use your idea, thank you